streams/jvm/src/test/scala/scalaz/zio/stream/StreamSpec.scala

```diff
@@ -529,4 +535,34 @@ class StreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
         }
     ) must_== Failure(Cause.Fail(e))
   }
+
+  private def bufferStream = prop { list: List[Int] =>
+    unsafeRunSync(
+      Stream
+        .fromIterable(list)
+        .buffer(2)
+        .run(Sink.collect[Int])
+    ) must_== (Success(list))
+  }
+
+  private def bufferStreamError = {
+    val e = new RuntimeException("boom")
+    unsafeRunSync(
+      (Stream.range(0, 10) ++ Stream.fail(e))
+        .buffer(2)
+        .run(Sink.collect[Int])
+    ) must_== Failure(Cause.Fail(e))
+  }
+
+  private def fastProducerSlowConsumer =
+    unsafeRun(
+      for {
+        ref          <- Ref.make(List[Int](1, 10))
```

Comment:
This is the general direction, so that’s good, but the test is not asserting on the right thing. The purpose is to use the ref as an indicator that the stream has executed its effects up to the `buffer` stage.

So you want the stream to look something like this:
```
_ <- Stream.fromIterable(List(1, 2, 3)).mapM(el => ref.update(el :: _)).buffer(2).mapM(_ => IO.never).drain.fork
```

What would happen here (theoretically) is that the `buffer` will allow the elements to make it through the first `mapM` and go into the ref, but they won’t make it to the end of the stream because of the `IO.never`.

So your `ref` becomes the indication that the elements are indeed being buffered.

Make sense?

Reply:
@iravid Thank you for the review. It makes sense. Fixed!. Kindly review.